CarpetX/src/driver.cxx

```diff
@@ -1669,14 +1673,15 @@ void SetupGlobals() {
     }
 
     // Allocate data
+    const size_t factor = group.vartype == CCTK_VARIABLE_COMPLEX ? 2 : 1;
     const nan_handling_t nan_handling = arraygroupdata.do_checkpoint
                                             ? nan_handling_t::forbid_nans
                                             : nan_handling_t::allow_nans;
     arraygroupdata.data.resize(group.numtimelevels);
     arraygroupdata.valid.resize(group.numtimelevels);
     for (int tl = 0; tl < int(arraygroupdata.data.size()); ++tl) {
       // TODO: Allocate in managed memory
-      arraygroupdata.data.at(tl).resize(arraygroupdata.numvars *
+      arraygroupdata.data.at(tl).resize(factor * arraygroupdata.numvars *
```

Comment:
I think we need to take `factor` into account also when calculating the pointers to the individual variables.

Reply:
Where did we calculate the pointers? ( I modified it a bit base on my guessing :) )